tensorflow/tools/dockerfiles/partials/ubuntu-devel.partial.Dockerfile

```diff
@@ -0,0 +1,24 @@
+ARG UBUNTU_VERSION=16.04
+FROM ubuntu:${UBUNTU_VERSION}
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        build-essential \
+        curl \
+        git \
+        libcurl3-dev \
+        libfreetype6-dev \
+        libhdf5-serial-dev \
+        libpng12-dev \
+        libzmq3-dev \
+        pkg-config \
+        python-dev \
+        rsync \
+        software-properties-common \
+        unzip \
+        zip \
+        zlib1g-dev \
+        openjdk-8-jdk \
+        openjdk-8-jre-headless \
+        && \
```

Comment:
Probably want to label this ubuntu1604-devel.partial.Dockerfile because the package list and names will change with 18.04.

Reply:
That's a great idea, but I think relabeling should come later, after this design is implemented. The set of images here aren't meant to be perfect; just a re-implementation of the images we currently have.

This suggestion would be an excellent first fix for someone other than me (either an internal team member or an OSS developer!) to confirm that working with the design is easy to do.